async_processor/types.py

```diff
@@ -12,17 +12,69 @@ class ProcessStatus(str, enum.Enum):
     SUCCESS = "SUCCESS"
 
 
-class InputMessage(BaseModel):
+class InputMessageBase(abc.ABC, BaseModel):
+    @abc.abstractmethod
+    def get_request_id(self) -> Optional[str]:
+        ...
+
+    @abc.abstractmethod
+    def get_published_at_epoch_ns(self) -> Optional[int]:
+        ...
+
+    @abc.abstractmethod
+    def get_body(self) -> Any:
+        ...
+
+
+class InputMessage(InputMessageBase):
     request_id: constr(regex=r"^[a-zA-Z0-9\-]{1,36}$")
     body: Any
     published_at_epoch_ns: Optional[int] = None
 
+    class Config:
+        extra = Extra.forbid
+
+    def get_request_id(self) -> Optional[str]:
+        return self.request_id
+
+    def get_published_at_epoch_ns(self) -> Optional[int]:
+        return self.published_at_epoch_ns
+
+    def get_body(self) -> Any:
+        return self.body
+
+
+# We cannot maintain two different types and should remove `InputMessage`
+# after sometime
+class InputMessageV2(InputMessageBase):
+    tfy_request_id: Optional[constr(regex=r"^[a-zA-Z0-9\-]{1,36}$")] = None
+    tfy_published_at_epoch_ns: Optional[int] = None
```

Comment:
Just for my understanding,
In this case the service enqueuing is responsible for setting these in the payload right?
- Without tfy_request_id they can't specifically subscribe/look for corresponding output message
- Without tfy_published_at_epoch_ns they lose out on queue wait time telemetry
Antyhing else?

Reply:
> In this case the service enqueuing is responsible for setting these in the payload right?

Yes

Reply:
> Without tfy_request_id they can't specifically subscribe/look for corresponding output message

Yes. If they are using NATS as output.

Reply:
> Without tfy_published_at_epoch_ns they lose out on queue wait time telemetry

Yes.